ui/src/app/pages/data/concept/concept-add-modal.spec.tsx

```diff
@@ -58,78 +66,51 @@ describe('ConceptAddModal', () => {
   });
 
   it('displays option to add to existing concept set if concept set in domain exists', async () => {
-    const wrapper = component();
+    component();
     const stubSetsInDomain = conceptSetsApi.conceptSets
       .filter((s) => s.domain === activeDomainTab.domain)
       .map((s) => s.name);
-    await waitOneTickAndUpdate(wrapper);
-    expect(
-      wrapper.find('[data-test-id="add-to-existing"]').exists()
-    ).toBeTruthy();
-    const foundSets = wrapper
-      .find('[data-test-id="existing-set"]')
-      .map((s) => s.text());
+    expect(await screen.findByTestId('add-to-existing')).toBeInTheDocument();
+    const foundSets = (await screen.findAllByTestId('existing-set')).map(
+      (s) => s.textContent
+    );
     expect(foundSets).toEqual(stubSetsInDomain);
   });
 
   it('disables option to add to existing if concept set does not exist & defaults to create', async () => {
     props.activeDomainTab = CardCountStubVariables.STUB_CARD_COUNTS[2];
-    const wrapper = component();
-    await waitOneTickAndUpdate(wrapper);
-    expect(
-      wrapper.find('[data-test-id="add-to-existing"]').exists()
-    ).toBeFalsy();
-    expect(
-      wrapper.find('[data-test-id="create-new-set"]').exists()
-    ).toBeTruthy();
-    expect(
-      wrapper
-        .find('[data-test-id="toggle-existing-set"]')
-        .first()
-        .prop('disabled')
-    ).toBe(true);
+    component();
+    expect(screen.queryByTestId('add-to-existing')).not.toBeInTheDocument();
+    expect(await screen.findByTestId('create-new-set')).toBeInTheDocument();
```

Comment:
Good call. Rather than a waitFor, I switched the order. I believe that they should be testing against the same render-state this way.

I am hesitant to use waitFor here, because I am testing the non-existence of something and not its disappearance.

What do you think?

Reply:
great, this works